syft/execution/role.py

```diff
@@ -295,15 +293,24 @@ def detail(worker: AbstractWorker, role_tuple: "Role") -> tuple:
         for ph in state.state_placeholders:
             placeholders[ph.id.value] = ph
 
-        return Role(
+        role = Role(
             id=id_,
             owner=worker,
             actions=actions,
             state=state,
-            placeholders=placeholders,
+            # placeholders=placeholders,
```

Comment:
It might still be okay to pass the `placeholders` here. Should be okay to modify the `placeholders` after they're set on the `Role`, yeah?